exchanges/tolerance_test.go

```diff
@@ -0,0 +1,260 @@
+	if !errors.Is(err, nil) {
+		t.Fatalf("expected error %v but received %v", nil, err)
+	}
+
+	newLimits := []MinMaxLevel{
+		{
+			Pair:      btcusd,
+			Asset:     asset.Spot,
+			MinPrice:  100000,
+			MaxPrice:  1000000,
+			MinAmount: 1,
+			MaxAmount: 10,
+		},
+	}
+
+	err = e.LoadTolerances(newLimits)
+	if !errors.Is(err, nil) {
+		t.Fatalf("expected error %v but received %v", errCannotLoadTolerance, err)
+	}
+
+	err = e.CheckTolerance(asset.Futures, ltcusd, 1337, 1337)
+	if !errors.Is(err, errCannotValidateAsset) {
+		t.Fatalf("expected error %v but received %v", errCannotValidateAsset, err)
+	}
+
+	err = e.CheckTolerance(asset.Spot, ltcusd, 1337, 1337)
+	if !errors.Is(err, errCannotValidateBaseCurrency) {
+		t.Fatalf("expected error %v but received %v", errCannotValidateBaseCurrency, err)
+	}
+
+	err = e.CheckTolerance(asset.Spot, btcltc, 1337, 1337)
+	if !errors.Is(err, errCannotValidateQuoteCurrency) {
+		t.Fatalf("expected error %v but received %v", errCannotValidateQuoteCurrency, err)
+	}
+
+	err = e.CheckTolerance(asset.Spot, btcusd, 1337, 1337)
+	if !errors.Is(err, ErrPriceExceedsMin) {
+		t.Fatalf("expected error %v but received %v", ErrPriceExceedsMin, err)
+	}
+
+	err = e.CheckTolerance(asset.Spot, btcusd, 1000001, 1337)
+	if !errors.Is(err, ErrPriceExceedsMax) {
+		t.Fatalf("expected error %v but received %v", ErrPriceExceedsMax, err)
+	}
+
+	err = e.CheckTolerance(asset.Spot, btcusd, 999999, .5)
+	if !errors.Is(err, ErrAmountExceedsMin) {
+		t.Fatalf("expected error %v but received %v", ErrAmountExceedsMin, err)
+	}
+
+	err = e.CheckTolerance(asset.Spot, btcusd, 999999, 11)
+	if !errors.Is(err, ErrAmountExceedsMax) {
+		t.Fatalf("expected error %v but received %v", ErrAmountExceedsMax, err)
+	}
+
+	err = e.CheckTolerance(asset.Spot, btcusd, 999999, 7)
+	if !errors.Is(err, nil) {
+		t.Fatalf("expected error %v but received %v", nil, err)
+	}
+}
+
+func TestConforms(t *testing.T) {
+	var tt *Tolerance
+	err := tt.Conforms(0, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tt = &Tolerance{
+		minNotional: 100,
+	}
+
+	err = tt.Conforms(1, 1)
+	if !errors.Is(err, ErrNotionalValue) {
+		t.Fatalf("expected error %v but received %v", ErrNotionalValue, err)
+	}
+
+	err = tt.Conforms(200, .5)
+	if !errors.Is(err, nil) {
+		t.Fatalf("expected error %v but received %v", nil, err)
+	}
+
+	tt.stepSizePrice = 0.001
+	err = tt.Conforms(200.0001, .5)
+	if !errors.Is(err, ErrPriceExceedsStep) {
+		t.Fatalf("expected error %v but received %v", ErrPriceExceedsStep, err)
+	}
+	err = tt.Conforms(200.004, .5)
+	if !errors.Is(err, nil) {
+		t.Fatalf("expected error %v but received %v", nil, err)
+	}
+
+	tt.stepSizeAmount = 0.001
+	err = tt.Conforms(200, .0002)
+	if !errors.Is(err, ErrAmountExceedsStep) {
+		t.Fatalf("expected error %v but received %v", ErrAmountExceedsStep, err)
+	}
+	err = tt.Conforms(200000, .003)
+	if !errors.Is(err, nil) {
+		t.Fatalf("expected error %v but received %v", nil, err)
+	}
+}
+
+func TestConformToAmount(t *testing.T) {
```

Comment:
Maybe we should add in a test case where amount is smaller than stepsize just to see if it returns 0